vsphere/host_port_group_structure.go

```diff
@@ -0,0 +1,141 @@
+package vsphere
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const hostPortGroupIDPrefix = "tf-HostPortGroup"
+
+// schemaHostPortGroupSpec returns schema items for resources that
+// need to work with HostPortGroupSpec, such as port groups.
+func schemaHostPortGroupSpec() map[string]*schema.Schema {
+	s := map[string]*schema.Schema{
+		// HostPortGroupSpec
+		"name": &schema.Schema{
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The name of the port group.",
+			ForceNew:    true,
+		},
+		"vlan_id": &schema.Schema{
```

Comment:
Thoughts on adding `ValidateFunc: validation.IntBetween(0, 4095),`

Reply:
Will add it!